opengever/maintenance/scripts/fix_retention_period.py

```diff
@@ -0,0 +1,465 @@
+        repo_folders = set(each.getObject() for each in repo_folder_brains)
+
+        logger.info(20*'-')
+        logger.info("{} of {} repository folders have been checked/fixed"
+                    .format(len(self.fixed_folders), len(repo_folders)))
+
+        omitted_folders = repo_folders.difference(self.fixed_folders)
+        omitted_folders = sorted(omitted_folders,
+                                 key=lambda content: content.absolute_url())
+        logger.info('The following repository folders have not been '
+                    'checked/fixed:')
+        for omitted in omitted_folders:
+            logger.info("{} {}".format(
+                omitted.Title(), '/'.join(omitted.getPhysicalPath())))
+
+    def get_repository_file_path(self):
+        profile_info = self.portal_setup.getProfileInfo(self.profile)
+        profile_path = profile_info['path']
+
+        repositories_folder = os.path.join(profile_path,
+                                           REPOSITORIES_FOLDER_NAME)
+
+        def is_parsable(filename):
+            if filename.startswith('.') or filename.startswith('~'):
+                return False
+            if not filename.endswith('.xlsx'):
+                return False
+            return True
+
+        repository_filenames = [filename for filename in
+                                os.listdir(repositories_folder)
+                                if is_parsable(filename)]
+
+        if len(repository_filenames) != 1:
+            raise Abort("Expected one repository file but found {}, {}".format(
+                        len(repository_filenames), repository_filenames))
+        return os.path.join(repositories_folder, repository_filenames[0])
+
+    def get_repo_root(self):
+        repo_roots = self.catalog.unrestrictedSearchResults(
+            portal_type={'query': 'opengever.repository.repositoryroot',
+                         'depth': 1})
+        if len(repo_roots) != 1:
+            raise Abort('Expected exactly one repository root, found {}'
+                        .format(len(repo_roots)))
+
+        return repo_roots[0].getObject()
+
+
+def main():
+    app = setup_app()
+    parser = setup_option_parser()
+    parser.add_option("-n", dest="dry_run", action="store_true", default=False)
+    parser.add_option("-p", dest="profile",
+                      help="profile that contains the repository excel file.")
+    options, args = parser.parse_args()
+
+    if not options.profile:
+        logger.error("the profile (-p) argument is required.")
+        return
+    if ":" not in options.profile:
+        logger.error("invalid profile id: '{}', missing ':'"
+                     .format(options.profile))
+        return
+
+    if options.dry_run:
+        logger.warn('transaction doomed because we are in dry-mode.')
+        transaction.doom()
+
+    plone = setup_plone(app, options)
+    RetentionPeriodFixer(plone, options).run()
+    transaction.commit()
```

Comment:
Instead, a big fat warning at the bottom that the output above is from a dry-run would be good to have.